### DISCUSSION TITLE
Does video shortcode not support assets directory?

### DISCUSSION THREAD

**Discussion by kugarocks** [thuliteio/doks#1295] [Q&A] [answered]
Hello, I want to reference a video in the assets directory. This is my shortcode.

```markdown
{{< video src="videos/misc/say-my-name.mp4" controls="true" >}}
```

But the error comes out.

```txt
ERROR No file with filename "videos/misc/say-my-name.mp4" found.
```

So I check the file `node_modules/@thulite/doks-core/layouts/shortcodes/video.html`.

```html
{{- $video_files := (.Page.Resources.Match (printf "%s*" $video_src)) -}}
```

It seems that it only support page resources.
When I use `resources.Match` instead of `.Page.Resources.Match`, it works.
Would it be better to use both methods for resources like this.

```html
  {{- $global_video_files := (resources.Match (printf "%s*" $video_src)) -}}
  {{- $video_files := (.Page.Resources.Match (printf "%s*" $video_src)) -}}
  {{- $video_files = $video_files | append $global_video_files -}}
```

What is the best practice to do that? I'm not sure if I should retrieve it from `resources`. Thanks for your help!

**Comment by h-enk** [comment, marked as answer]
I think it should support page resources, global resources, and remote resources (align with [Resource Locations](https://images.thulite.io/docs/basics/resources/) for images).

So, using [this logic](https://github.com/thuliteio/images/blob/main/layouts/partials/img.html#L244-L283)

**Reply by kugarocks** [reply]
Thanks a lot.
